src/Modal.tsx

```diff
@@ -0,0 +1,114 @@
+/** @jsx jsx */
+import { jsx, css } from "@emotion/core";
+import React, { useEffect } from "react";
+import * as typography from "./typography";
+import * as colors from "./colors";
+
+interface Props {
+    size: "small" | "standard" | "large";
+    title: string;
+    description?: string;
+    children: React.ReactElement;
+    onClose: (() => void);
+}
+
+const modalBackdrop = css`
+        position: fixed;
+        top: 0px;
+        left: 0px;
+        width: 100%;
+        height: 100%;
+        
+        &:after {
+            position: fixed;
+            z-index: 10;
+            content: '';
+            background-color: ${colors.grey.lighter};
+            opacity: 0.7;
+            top: 0;
+            left: 0;
+            bottom: 0;
+            right: 0;
+            z-index: -1;
+        }
+    `;
+
+const ESCAPE_KEY_CODE = 27;
+
+const getTitleTypography = ( size: Props["size"]) => {
+    if(size === "small") {
+        return {...typography.base.large,
+                fontWeight: 600};
+    }
+    return {...typography.base.xxlarge};
+}
+
+const getModalWidth = ( size: Props["size"]) => {
```

Comment:
(non-blocking)  

Curious how you feel about

```tsx
const modalWidths = {
  small: 460,
  standard: 600,
  large: 800,
}

// ...

modalWidths[props.size]

```

Reply:
```
modalWidths: Record<React.ComponentProps<Modal>["size"], CSS.WidthProperty> = {
  small: 460,
  standard: 600,
  large: 800,
}
```

I ended up changing this to use a `switch` statement with the `default` of `assertUnreachable(size)`, in which `size` is `:never`, but I think your way is way simpler.